Add translation strings

ktohalloran opened the pull request:
## Overview

This PR adds translation strings that switch when the translation buttons are clicked. 

### Demo

![Screen Shot 2021-12-13 at 11 16 46 AM](https://user-images.githubusercontent.com/77936689/145848427-e37cea36-d30c-4623-9045-271a580689b1.png)

Translated into Spanish

![Screen Shot 2021-12-13 at 11 16 55 AM](https://user-images.githubusercontent.com/77936689/145848467-303b6947-e880-49aa-b8fd-0426305ec9f8.png)

Translated in Chinese

### Notes

While some of the strings were very straight forward, a lot of them were not. Here is an overview of the problems faced and the decisions made. 

**Missing Translations**
* Missing translations in **Spanish**: "Peak", "Off Peak" (in dock when commuter trains are selected in profile), "Site by" (dock)
* Missing translations in **Chinese**: "New Search" (nav bar), "when" (dock), "Apartments" (nav bar), "Resend invitation email" (edit profile), "Primary" (edit profile)
* Missing translations in **both languages**: "Peak No Express", "Off Peak No Express" (in dock when commuter trains are _not_ selected in profile), "of" (in the dock, e.g. 1-3 of 142), all neighborhood descriptions

**Amplify Code**

There were also elements that had translations but were not incorporated for similar reasons as laid out in [PR 377](https://github.com/azavea/echo-locator/pull/377). As quoted from that PR, the log in components, especially "Sign Out", "Username", "Password", "Forgot your password?", and "Reset password" are not translated "because [Amplify code] apparently uses its own, separate translation machinery. This primarily affects the login flow, and the translation switcher isn't available on the login page anyway, so I didn't try to find a solution. In any event, we won't be able to use Amplify's login components once we switch to Django authentication, so eventually the login components will be migrated to `react-i18next` too."

**Grammar in Dynamic Phrases**

One of the issues I ran into with translating dynamic phrases into Chinese was how we structured sentences. For example, the phrase "Take [bus number] in [number] minutes" in the dock is currently structured with "Take", "in", and "minutes" as separate phrases that get translated separately and pieced together into a sentence along with the dynamic values within the component. There are grammatical issues with some of the existing translations, as well, especially in phrases that have been structured piece-meal. One example is in the "[mode of transportation] from [destination]" phrase in the dock. This particular sentence has grammatical issues at least in Spanish and possibly also in Chinese, again due to how we have chosen to structure that component. These sentences also have an added complexity in that the phrases used to complete the mode of transportation and destination name are connected to other pieces of the app, particularly the Profile page, where one version of the word might make sense but then does not when used in a sentence in another piece of the app. As an example, the app uses the Spanish word "trabajo" in the Profile page destination dropdown menu but that word should become "el trabajo" when used in a sentence and, if the word before it is "de", should become "del trabajo."  A separate task (#389) has been added to explore alternate ways of structuring the problematic phrases to allow for more flexible grammatical rules. 

**Peak/Off Peak/Peak No Express/Off Peak No Express Dropdown Menu**

The Peak/Off Peak/Peak No Express/Off Peak No Express Dropdown Menu has not been translated, as I ran into an issue with how the `Select` element in `form.js` generates the options and sends those options to network state. I tried a few different methods, such as adding the translate method into a mapping function that generates the network options to the Select component and translating within the Select function as the options are generated; however, there is a disconnect between how the option labels need to look to work with `i18next` and the option labels that state expects. Ultimately, for the purpose of this PR, I decided that this should be set aside as its own future task (#391) because we currently do not have translations for any of the options in Spanish and for two of the options in Chinese.

**Maintaining State Across Multiple Languages**

As of right now, the destination label in the "Show Travel Times: from" dropdown menu is always in the language that it was set in from the Profile page and not in the language that is currently active. This hints at a larger issue as to how state understands the same value but in a different language that may be related to the Peak/Off Peak menu issue related above; however, I'm not sure this particular manifestation is a huge issue, as in theory users will not be going back and forth between languages.

## Testing Instructions

 * Either run `./scripts/server` or `cd taui` &#8594; `yarn start`
 * Click on the "Español" button. Check the dock, neighborhood detail view, marker pop ups, and edit profile page to verify that any English has been addressed above.
 * Click on the "中文" button. Check the dock, neighborhood detail view, marker pop ups, and edit profile page to verify that any English has been addressed above.


Resolves #384

Commit:
Translate static strings into Spanish

This does not include the static strings on the map popups as that
PR had not been merged at the time of this commit; those will be
added later.

Commit:
Translate dynamic strings into Spanish

Commit:
Translate Chinese static strings

Commit:
Translate some dynamic strings into Chinese

Many of these more dynamic strings require an alteration of the
order of the sentence from Spanish and English. While this was
easy with how we created the text for the tooltips on the meters
in the neighborhood details, it was not as easy for elements such
as the "About [number] minutes via bus from [location]" or the
"Take [transit route] in [number] minutes". These will require
a restructuring of those elements to make the grammar more
flexible, as well as clearer translations from the client so
that the developer does not have to guess as to the order and
proper grammar.

Commit:
Translate content on map pop ups

The Chinese and Spanish translations for the "Click marker for
details" are not up-to-date; they currently translate to "Click
for Listing". This commit also deconstructs the `beds` string
within data in order to translate the abbrevation for bedrooms.

KlaasH commented on taui/src/locales/zh/translations.js:
```diff
@@ -109,89 +115,88 @@ export default {
+    AboveAverage: '高於平均水平',
+    Average: '相對平均水平',
+    BelowAverage: '低於平均水平',
+    Education: '<p>在參考過學校的整體質量和低收入家庭學生的表現，（如有低收入家庭學生表現的數據可使用），<strong>{{town}}</strong> 的學校在本州 <strong>{{averageRelation}}</strong> ，其排名高於[ ]的其他學區 <strong>{{edPercentile}}</strong> (資料來源: <a href="http://profiles.doe.mass.edu/" target="_blank">馬薩諸塞州中小學教育部 MA DESE</a>)</p>',
```
This happened to catch my eye, and it doesn't really make sense that the Chinese translation of "DOE" would be "DESE".  It looks like despite the URL being `doe`, the department is called the Department of Elementary and Secondary Education and the site uses DESE.  The source strings for the Chinese translation must have been pulled at a time when the English text said DESE, then it must have been changed later.
It kind of seems like it should be changed to DESE in all three translations files, though there might be more to the situation since I don't know why it would have been changed to DOE.

ktohalloran replied:
Yeah, I also noticed this and thought it was strange so thanks for catching it. I also agree it should be changed to DESE because the app uses DOE to refer to the US Dept of Energy in other places.

KlaasH reviewed (approved):
This looks good.  Clearly there are a number of lingering issues around translation, but the approach of focusing on the task at hand (getting the strings we have into the translations file) and spinning out the others makes sense.  

I think the last one you mentioned--the translated word in the "Show Travel Times: from" droplist not changing with the language selection--also deserves an issue (or to be added to #391, since they're right next to each other and might be similar enough that it would make sense to do them together).  There must be a way to reconfigure how that's built to make it respond to the selected language, and I don't think we should leave a case of mixed translation without at least having an issue about it.

There's one line comment below, which is just kind of a weird detail. Not sure if it's even worth making a thing out of, though on the other hand we could also avoid making a thing of it by just changing them all to DESE and calling it good.

So either way on that, and either a new issue or a revision to #391 for the "from" droplist, then this is good to go :+1:

Commit:
Change source MA DOE to MA DESE in education tooltips

The Chinese, English and Spanish translations disagreed about
whether the Massachusetts Department of Elementary and
Seconary Education was cited as "MA DOE" or "MA DESE". This commit
changes "DOE" in the English and Spanish translation files to
match the Chinese file in saying "DESE", as "DOE" is used
elsewhere in the app to refer to the US Dept of Energy.
